File: clrs/cpp/clrs/part7/chap32/string_matching_test.cpp
#include "string_matching.h"
#include<iostream>
using namespace std;
using namespace part7::chap32;

int main(){

    auto idx = StringMatching::RabinKarp(
        "TEST",
        "THIS IS A TEST TEXT"
    );
    
    cout<<" Runing test "<<idx;
    return !(idx == 10);
}

File: clrs/csharp/CLRS.Tests/StringMatchingTest.cs
namespace CLRS.Tests;
using CLRS.Part7.Chap32;

public class StringMatchingTest
{
    [Fact]
    public void TestRabinKarp()
    {
         //Assert.Equal(10,StringMatching.RabinKarp("TEST","THIS IS A TEST TEXT"));
    }

    [Fact]
    public void TestKnuthMorisPratt(){
        // Assert.True(StringMatching.KnuthMorrisPratt("nitin","nitin") == 0,"shoudl be true");
    }

     [Fact]
    public void TestKnuthMorisPrattMultipleText(){
         var searcher = new KMPSearch("abcabcacab");
       //  var first = searcher.Search("babcbabcabcaabcabcabcacabc");
         var first = searcher.Search("babcbabcabcbabcabcabcacabc");
         Assert.Equal(15, first);
    }
}
